Homework 6 (JDBC)

sergavv14 opened the pull request.

sergavv14 committed:
Merge pull request #1 from pesua/master

Spring IoC homework

sergavv14 committed:
Merge https://github.com/pesua/skillsup_java1

sergavv14 committed:
Save classwork jdbs

sergavv14 committed:
Fixed erorr in request "Select ... INNER JOIN ..."

sergavv14 committed:
Write queries that will return - Done

pesua commented on src/test/java/ua/dp/skillsup/jdbc/JdbcTest.java:
```diff
@@ -37,14 +37,315 @@ public void testUser() throws Exception {
+                "t2.TITLE, " +
+                "t1.USERNAME, " +
+                "t2.TIMESTAMP " +
+                "FROM USER t1 INNER JOIN POST t2 " +
```
Тут применимы те же советы что и к обычному коду, давай читаемые имена. select p.title, u.username итд понимать гораздо легче.

pesua commented on src/test/java/ua/dp/skillsup/jdbc/JdbcTest.java:
```diff
@@ -37,14 +37,315 @@ public void testUser() throws Exception {
+                "t1.USERNAME, " +
+                "t2.TIMESTAMP " +
+                "FROM USER t1 INNER JOIN POST t2 " +
+                "ON (t1.ID = t2.AUTHOR_ID)").executeQuery();
```
Скобки тут не обязателньные, не видел чтоб кто-то писал.

pesua commented on src/test/java/ua/dp/skillsup/jdbc/JdbcTest.java:
```diff
@@ -37,14 +37,315 @@ public void testUser() throws Exception {
+
+        /*
+        ________
+        a) list of all post with columns (title, username)
```
Спасибо за заботу, но лучше ревьюер скачает и запустит чем мы будем хранить распухший от коментариев код).

pesua commented on src/test/java/ua/dp/skillsup/jdbc/JdbcTest.java:
```diff
@@ -37,14 +37,315 @@ public void testUser() throws Exception {
+                "ON (t1.POST_ID = t2.POST_ID) " +
+                "GROUP BY t1.POST_ID";
+
+        ResultSet resultSet5 = conn.prepareStatement("SELECT " +
```
Тут еще можно обойтись без подзапроса.

pesua commented on src/test/java/ua/dp/skillsup/jdbc/JdbcTest.java:
```diff
@@ -37,14 +37,315 @@ public void testUser() throws Exception {
+        statement.execute("DROP TABLE IF EXISTS POST;" +
+                "CREATE TABLE POST\n" +
+                "(\n" +
+                "    POST_ID BIGINT NOT NULL AUTO_INCREMENT PRIMARY KEY,\n" +
```
Обычно первичный ключ обызвают просто ID, POST_ID выглядит как foreign key.

pesua commented:
![image](https://user-images.githubusercontent.com/989360/35846592-f3451b88-0b1f-11e8-9b38-47f4144c441a.png)

sergavv14 committed:
Fixed all remarks:
1) Give readable names to tables - Done
2) Del some parentheses in the query - Done
3) Del comment - Done
4) Add new version of the query "E.2." - Done
5) Rename the primary key POST_ID => ID in table POST - Done

pesua commented on src/test/java/ua/dp/skillsup/jdbc/JdbcTest.java:
```diff
@@ -37,14 +37,243 @@ public void testUser() throws Exception {
+
+        System.out.println("");
+        System.out.println("Version E.2 :");
+        ResultSet resultSet8 = conn.prepareStatement("SELECT " +
```
:+1: можно было старую версию не оставлять

pesua reviewed (commented):
Отлично. Сорян что так долго проверял.

pesua commented:
![image](https://user-images.githubusercontent.com/989360/36309638-1f968a9e-132e-11e8-9040-6b08c52fd55f.png)
